inc/az_ulib_descriptor_api.h

```diff
@@ -0,0 +1,210 @@
+    const az_ulib_action_set set;
+    const az_ulib_action_cancelation_callback cancel;
+  } action_ptr_2;
+  const uint8_t flags; /**<This is a 8 bit flags that handles internal status of the action. */
+} az_ulib_action_descriptor;
+
+/**
+ * @brief   Interface descriptor.
+ *
+ * This structure contains the full information about a collection of action that represents the
+ * interface.
+ */
+typedef struct az_ulib_interface_descriptor_tag {
+  const char *name; /**<The `\0` terminated <tt>const char*</tt> with the interface name. */
+  az_ulib_version version; /**<The #az_ulib_version with the interface version. */
+  uint8_t size; /**<The <tt>uint8_t</tt> with the number of actions in the interface. */
+  az_ulib_action_descriptor *action_list; /**<The list of #az_ulib_action_descriptor with the
+                                          actions in this interface. */
+} az_ulib_interface_descriptor;
+
+/**
+ * @brief   Create a new Interface descriptor.
+ *
+ * This API creates a new description for a interface, it will create the
+ * #az_ulib_interface_descriptor with the provided interface name and version, adding all the
+ * actions.
+ *
+ * The descriptor will be stored in the Text area (Flash) that will be pointed by the provided
+ * interface_var. On this way, no other components on the system needs to copy any of the data on
+ * the descriptor.
+ *
+ * @param[in]   interface_var   The <tt>az_ulib_interface_descriptor</tt> the will pointe to the
+ *                              created descriptor.
+ * @param[in]   interface_name  The `/0` terminated <tt>const char* const</tt> with the interface
+ *                              name. It cannot be <tt>NULL</tt> and shall be allocated in a way
+ *                              that it stays valid until the interface is unpublished at some
+ *                              (potentially) unknown time in the future.
+ * @param[in]   version         The <tt>az_ulib_version</tt> with the interface version.
+ * @param[in]   ...             The list of <tt>az_ulib_action_descriptor</tt> with the actions in
+ *                              the interface.
+ */
+#define AZ_ULIB_DESCRIPTOR_CREATE(interface_var, interface_name, version, ...) \
+  static const az_ulib_action_descriptor MU_C2(interface_var, _ACTION_LIST)[] \
+      = { MU_FOR_EACH_1(MU_DEFINE_ENUMERATION_CONSTANT, __VA_ARGS__) }; \
+  static const az_ulib_interface_descriptor MU_C1(interface_var) \
+      = { (interface_name), \
+          (version), \
+          (uint8_t)(MU_COUNT_ARG(__VA_ARGS__) / 4), \
+          (az_ulib_action_descriptor *)MU_C2(interface_var, _ACTION_LIST) };
+
+/**
+ * @brief   Add property to the interface descriptor.
+ *
+ * Populate a new `property` action descriptor to add to the interface. On the interface context, a
+ * `property` is a data exposed by its `get` and `set` function.
+ *
+ * @param[in]   name        The `/0` terminated <tt>const char* const</tt> with the property name.
```

Comment:
><tt> [](start = 48, length = 4)

with that being said these should likely be changed to a ` instead. I can do a quick find all and replace after you push yours though since it will change files not within the scope of this PR. It'll be easier to do it after than you manually changing just your files